core/src/test/java/org/apache/iceberg/avro/AvroTestHelpers.java

```diff
@@ -126,9 +139,18 @@ private static void assertEquals(Type type, Object expected, Object actual) {
         assertThat(actual).as("Primitive value should be equal to expected").isEqualTo(expected);
         break;
       case STRUCT:
-        assertThat(expected).as("Expected should be a Record").isInstanceOf(Record.class);
-        assertThat(actual).as("Actual should be a Record").isInstanceOf(Record.class);
-        assertEquals(type.asStructType(), (Record) expected, (Record) actual);
+        assertThat(expected)
+            .as("Expected should be a Record or StructLike")
+            .isInstanceOfAny(Record.class, StructLike.class);
+
+        if (expected instanceof Record) {
+          assertThat(actual).as("Actual should be a Record").isInstanceOf(Record.class);
+          assertEquals(type.asStructType(), (Record) expected, (Record) actual);
+        } else {
+          assertThat(actual).as("Actual should be a GenericRecord").isInstanceOf(StructLike.class);
```

Comment:
The message doesn't match the assertion. I think this should be `"Actual should be a StructLike"`.